crates/factors-test/src/lib.rs

```diff
@@ -27,60 +40,54 @@ impl Default for TestEnvironment {
             [component.empty]
             source = "does-not-exist.wasm"
         };
-        Self { manifest }
+        Self {
+            factors,
+            manifest,
+            runtime_config: Default::default(),
+        }
     }
-}
 
-impl TestEnvironment {
-    /// Builds a TestEnvironment by extending a default manifest with the given
-    /// manifest TOML.
+    /// Extends the manifest with the given TOML.
     ///
     /// The default manifest includes boilerplate like the
     /// `spin_manifest_version` and `[application]` section, so you typically
     /// need to pass only a `[component.test-component]` section.
-    pub fn default_manifest_extend(manifest_merge: toml::Table) -> Self {
-        let mut env = Self::default();
-        env.manifest.extend(manifest_merge);
-        env
+    pub fn extend_manifest(mut self, manifest_merge: toml::Table) -> Self {
```

Comment:
This seems to be mimicking a builder pattern. Normally this would be a separate struct with a `build` method, but I think in this case, just mutating the `TestEnvironment` is fine. Just wanted to note it here as a place for potential future change.